munit/shared/src/main/scala/munit/Assertions.scala

```diff
@@ -228,27 +226,61 @@ trait Assertions extends MacroCompat.CompileErrorMacro {
     }
   }
 
+  // for MIMA compatibility
+  @deprecated("Use version with implicit DiffOptions", "1.0.4")
+  protected def fail(
+      message: String,
+      cause: Throwable,
+      loc: Location,
+  ): Nothing = {
+    implicit val _loc: Location = loc
+    fail(message, cause)
+  }
+
   /**
    * Unconditionally fails this test with the given message and exception marked as the cause.
    */
-  def fail(message: String, cause: Throwable)(implicit loc: Location): Nothing =
-    throw new FailException(
-      munitFilterAnsi(munitLines.formatLine(loc, message)),
-      cause,
-      isStackTracesEnabled = true,
-      location = loc,
-    )
+  def fail(message: String, cause: Throwable)(implicit
+      loc: Location,
+      diffOptions: DiffOptions,
+  ): Nothing = throw new FailException(
+    munitLines.formatLine(loc, message, ansi = useAnsiColors),
+    cause,
+    isStackTracesEnabled = true,
+    location = loc,
+  )
+
+  // for MIMA compatibility
+  @deprecated("Use version with implicit DiffOptions", "1.0.4")
```

Comment:
I don't think it will work, though if someone is overriding the method it will show an error.

Reply:
this provides binary compatibility (the parameters, same order) except I'm not sure if private is sufficient, too; and the new method is source-compatible.

Reply:
Though if we override it wil fail the compilation or am I missing something:

```
trait Location
trait A{
  def locs(a: String, b: Int, l: Location): Nothing = ???
}

class B extends A{
  override def locs(a : String, b: Int)(implicit l : Location) = ???
}
```

This will fail, so I assume your change should also fail

Reply:
I would think so, which is why I am bit confused. It might be a bug in Mima or alternatively the code produced is binary compatible, just not source compatible.

Reply:
yes, mima checks for binary. well, if there's minor source incompatibility, increasing the minor version, as you suggested, should be enough. wdyt?